### PR TITLE
DEX-656 Timestamp in web socket messages

### PR DESCRIPTION
**Pull request by DenisSerebryansky**
* WsAddressState, WsOrderBook got timestamp
 * Diffx implicits that ignore timestamp added
 * WebSocketConnection logs any messages
 * WebSocketMessagesSerdeSpecification improved

### CODE REVIEW

File: dex-it-common/src/main/scala/com/wavesplatform/dex/it/api/websockets/HasWebSockets.scala [hunk 1]
```diff
@@ -8,16 +8,29 @@ import akka.actor.ActorSystem
...
 import org.scalatest.{BeforeAndAfterAll, Suite}
+import play.api.libs.json.Json
 
 trait HasWebSockets extends BeforeAndAfterAll { _: Suite =>
 
   implicit protected val system: ActorSystem        = ActorSystem()
   implicit protected val materializer: Materializer = Materializer.matFromSystem(system)
 
+  implicit protected val derivedAddressStateDiff: Diff[WsAddressState] = Derived {
+    getDiff[WsAddressState]((l, r) => l.balances == r.balances && l.orders == r.orders)
```
Comment (vsuharnikov) [inline dex-it-common/src/main/scala/com/wavesplatform/dex/it/api/websockets/HasWebSockets.scala]: Please use [ignore](https://github.com/softwaremill/diffx/blob/master/README.md#customization) and we will never forget to add a new field to comparison.

File: dex-it-common/src/main/scala/com/wavesplatform/dex/it/api/websockets/HasWebSockets.scala [hunk 2]
```diff
@@ -8,16 +8,29 @@ import akka.actor.ActorSystem
...
+  }
+
+  implicit protected val derivedOrderBookDiff: Diff[WsOrderBook] = Derived {
+    getDiff[WsOrderBook]((l, r) => l.asks == r.asks && l.bids == r.bids && l.lastTrade == r.lastTrade)
```
Comment (vsuharnikov) [inline dex-it-common/src/main/scala/com/wavesplatform/dex/it/api/websockets/HasWebSockets.scala]: Here too.

File: dex/src/test/scala/com/wavesplatform/dex/api/websockets/WebSocketMessagesSerdeSpecification.scala [hunk 3]
```diff
@@ -114,20 +95,38 @@ class WebSocketMessagesSerdeSpecification extends AnyFreeSpec with ScalaCheckDri
   private val wsOrderBookGen: Gen[WsOrderBook] = for {
     asks      <- wsSide(askPricesGen)
     bids      <- wsSide(bidPricesGen)
-    lastTrade <- Gen.oneOf[Option[WsLastTrade]](None, lastTrade.map(Option(_)))
-  } yield WsOrderBook(asks, bids, lastTrade)
+    lastTrade <- Gen.oneOf[Option[WsLastTrade]](None, lastTradeGen.map(Option(_)))
+  } yield WsOrderBook(asks, bids, lastTrade, System.currentTimeMillis)
```
Comment (vsuharnikov) [inline dex/src/test/scala/com/wavesplatform/dex/api/websockets/WebSocketMessagesSerdeSpecification.scala]:
```scala
ts <- Gen.choose(0L, Long.MaxValue)
```

here would be better ;)

File: dex-it-common/src/main/scala/com/wavesplatform/dex/it/api/websockets/HasWebSockets.scala (hunk 1 above)
Comment (DenisSerebryansky) [reply dex-it-common/src/main/scala/com/wavesplatform/dex/it/api/websockets/HasWebSockets.scala]: Done

File: dex-it-common/src/main/scala/com/wavesplatform/dex/it/api/websockets/HasWebSockets.scala (hunk 2 above)
Comment (DenisSerebryansky) [reply dex-it-common/src/main/scala/com/wavesplatform/dex/it/api/websockets/HasWebSockets.scala]: Done

File: dex/src/test/scala/com/wavesplatform/dex/api/websockets/WebSocketMessagesSerdeSpecification.scala (hunk 3 above)
Comment (DenisSerebryansky) [reply dex/src/test/scala/com/wavesplatform/dex/api/websockets/WebSocketMessagesSerdeSpecification.scala]: Done

**Pull request merged**